### DISCUSSION TITLE
Clear night on December 26, 2022

### DISCUSSION THREAD

**Discussion by WimDeMeester** [WimDeMeester/eFinder#48] [Show and tell (success stories and failures)]
When looking at the weather the last week, I did not expect a clear night anymore in 2022.  I was really surprised by a message of an observer if I was interested in observing at our local observing location.  
I only arrived at the site around 21:00, when already 6 other observers were enjoying the clear night.
I tested the eFinder for a while.  I made some screenshots of what was happening, but alas, I made the screenshots to the clipboard and not to the disk...  I can tell that the graticule was displayed correctly at the position of Polaris after I measured (and used) the offset, so all is fine here.
Also moving to other objects seemed to work quite fine.  I'm already happy with the improved accuracy of the automatic alignment with the eFinder.  Using the GoTo++ option also worked much better than during the previous tests.  The objects were not perfectly centered in the Field of View, but that might be because I did not use a high magnification for measuring the offset for polaris.  So, as far as I see now, everything seemed to work.  I don't know what I did different compared to previous time.  The telescope was on a more stable soil, but I don't think that can make a difference...

There were a few problems however...  I also tried the handpad only version, and I got an error (complaining about capture.axy... also here the screenshot is lost).  THis was in our updated version, so I will try to hunt down the problem.

Another problem was that after a while, it seemed that the connection with the Nexus DSC pro was lost... The Raspberry had still the correct IP address and as such could still see the Nexus, but nothing was sent from the Nexus to the Raspberry... Restartinh the eFinder software did not solve anything, I guess I needed to restart the Nexus to solve the problem.  @AstroKeith Do you have an idea what could have gone wrong here?

**Comment by AstroKeith** [comment]
Glad to hear things are getting better.

If the eFinder and goto++ are basically working, but not very accurate on the goto++ I would try a few things.

Yes, do the Polaris calibration with a reticule.

What is the eFinder saying the final delta is after goto++?

If it’s small, ie an arc minute or two, then if the target isn’t in the centre of the EP fov, then there is a real problem. Either eFinder code or the scope is flexing.

More likely, with ServoCat in my experience, the delta will be quite large, showing a backlash issue (or similar)

Have you changed the Nexus DSC Pro's SSID and password? Especially important is other Nexus DSC are within range. Also try to use channels 1,6 or 11 wifi on the Nexus. These dont overlap with other channels. Can you check if another wifi device os swamping the field? I have a wifi scanner app on my tablet to do just this.

Did you restart the eFinder code, or reboot the whole Pi? You should do the later in these cases.

Cheers
Keith

**Comment by WimDeMeester** [comment]
THanks for the answer!  I don't have a reticule, but I will do the polaris calibration with the highest magnification I have. 
I have changed the Nexus DSC Pro SSID and password, and I don't think any other Nexus users were at the observing site.  I think I rebooted the Pi, but I'm not sure anymore...

**Comment by AstroKeith** [comment]
I try not to buy Chinese, but this is a good option https://www.amazon.co.uk/Crosshair-Illuminated-Centering-Adjustable-Astrophotography/dp/B083LK45YY/ref=sr_1_4?crid=3I1XW3NSJWZ8H&keywords=svbony+reticle&qid=1672314777&sprefix=svbony+reticule%2Caps%2C62&sr=8-4

All my other eyepieces are Ethos, so I had to get a reticule! It’s good for checking as well as initial alignment.

The wifi antenna on the Pi is just a bit of tracking on the pcb. It’s up near the microSD card slot on the Pi4B. It can be seriously affected by nearby cables. Weirdly it shows as connected with a strong signal, but won’t transfer data correctly. This sounds a bit like your symptoms?

A few things to try, 
Make sure no cables are near that end. Especially USB3 types. They have a good earth screen and are very good at grounding wifi signals!
Make sure there isn’t much metal between the Pi and the Nexus.
Or, try a wifi dongle. If you get the right ones, the Pi will recognise it straight away and present it as WLAN1.

Cheers
Keith

**Comment by WimDeMeester** [comment]
Thanks for the link to the reticle eyepiece!  
I have a Wifi dongle attached to the Pi, so that should not be the problem...

**Comment by AstroKeith** [comment]
OK.

A few more checks!

I’ve just remembered yours is a Nexus ‘Pro’. So it has the wifi bug I think we have discussed. It will only reliably support one client. If you have SkySafari connected also, this is likely to crash the Nexus TCP socket. The SkySafari sends requests for position readouts between 4 and 10 Hz, so quite a load.

Interestingly, I was a SkySafari addict for years - until I added the eFinder. I was seemingly using the SKySafari as a reassurance display, making up for pointing errors, plus I needed it to display field stars when searching for difficult objects. With eFinder that all goes away. Also for many of the targets I want, the SkySafari database doesnt include them, or at least makes it difficult to display them. I’ve got into the habit of creating new catalogues for such classes and adding them to the Nexus SD card. (Serge gave me a copy of the utility needed to create the special binaries)

Using Channel 6 on the wifi seems to be better.

If you do get a crash, try turning wifi on and off on the Nexus to clear it.

Finally, are you sure the dongle is being used? The Pi and its OS is very fickle! Do you see two WLANs in ifconfig?

Cheers
Keith

**Reply by WimDeMeester** [reply]
I'm indeed using Channel 6 on the wifi.  There is only one WLAN in ifconfig, but I disabled the build-in wifi.  So yes, I'm sure I'm using the dongle.  I'll try to turn off and on the Wifi on the Nexus next time.  
Up to a few months ago, I had my Nexus attached to the bottom of the telescope, so SkySafari was really useful to go to the different objects.  I have upgraded my telescope with the [Markless DSC Stalk II](https://deepskywim.blogspot.com/2022/12/dsc-stalk.html), so now I have the Nexus at a very convenient place, so I guess I don't need SkySafari anymore.  I anyway have an Interstellarum Deep Sky atlas at hand and I can always just start up SkySafari if needed to check the field stars.  I guess I'll try to work without SkySafari next time.

**Comment by AstroKeith** [comment]
A couple of thoughts.

It is possible to use another wifi router to enable multiple client connections. For someone in the UK I used a TripMate self powered travel router (about £40), and configured the eFinder, Nexus and tablet to all connect in ‘infrastructure’ mode. You can then easily browse in with another computer to VNC the eFinder desktop.

I realise that the delta on the display, is 'solved' versus 'Nexus readout’. This is partly due to a hangover from previous days when I didnt have access to the goto target coordinates and had to assume that the scopedrive had done a good job of driving to the target and hence the Nexus readout was very close to the goto target coordinates.

We now have the actual goto coordinates (be they SkySafari or Nexus originated) and so the delta with respect to this is better I think. (Or have the option to display either)

A small revision. The GUI variant already has a function ‘ def readTarget()’

I’ll make the changes to my own variants, but leave it to you or Mike to work it into the next release on the repository?

Keith

**Reply by WimDeMeester** [reply]
Do I have to change a setting (from SkySafari to Nexus) in the settings file if I only use the Nexus?  I'll have a look if I can change to show to use the actual coordinates (I guess we need to use the coordinates given back by readTarget, if I understand it correctly).

**Comment by AstroKeith** [comment]
No change needed.

Serge modified his firmware such that when we ask for a target, we get whatever the last set was, regardless of how. If no target has been set yet we get, 00:00:00 which I trapped in the readTarget().

So we can give the user the choice of displaying delta with respect to Nexus readout ‘read_altAz()’ or wrt target, ‘readTarget{}. Probably the default should be wrt Target, unless not set yet, then wrt Nexus.

Keith

**Comment by WimDeMeester** [comment]
Sorry for being so quiet the last weeks.  I was on a technical mission to La Palma, so I could not test the eFinder.  Now, I'm just hoping for the clouds to go away...  I received the reticule eyepiece, so I can make sure that polaris is centered perfectly in the FoV next time.

**Comment by AstroKeith** [comment]
Excellent.
Just a reminder to make sure that on the Nexus, in settings/align, you have it set to '2 star w/o alt ref'
